src/overlays/actors/ovl_En_Am/z_en_am.c

```diff
@@ -89,81 +124,453 @@ static DamageTable D_808B10C8 = {
+    this->unk_240 = 6.0f;
+    this->actionFunc = func_808B03C0;
+}
+
+void func_808B03C0(EnAm* this, GlobalContext* globalCtx) {
+    this->unk_23C = this->actor.yawTowardsPlayer;
+    func_808B0208(this, globalCtx);
+    if ((this->actor.bgCheckFlags & 2)) {
+        this->unk_236--;
+    }
+    if (this->unk_236 == 0) {
+        func_808B0640(this);
+    } else if ((this->returnHomeTimer == 0) || Actor_XZDistanceToPoint(&this->actor, &this->actor.home.pos) > 240.0f) {
+        func_808B0460(this);
+    }
+}
+
+void func_808B0460(EnAm* this) {
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->unk_240 = 0.0f;
+    this->unk_23C = Actor_YawToPoint(&this->actor, &this->actor.home.pos);
+    this->actionFunc = func_808B04A8;
+}
+
+void func_808B04A8(EnAm* this, GlobalContext* globalCtx) {
+    func_808B0208(this, globalCtx);
+    if (this->unk_23C == this->actor.world.rot.y) {
+        func_808B057C(this);
+    }
+}
+
+void func_808B04E4(EnAm* this) {
+    this->unk_240 = 0.0f;
+    this->unk_23C = this->actor.home.rot.y;
+    this->actionFunc = func_808B0508;
+}
+
+void func_808B0508(EnAm* this, GlobalContext* globalCtx) {
+    func_808B0208(this, globalCtx);
+    if ((this->actor.bgCheckFlags & 1) == 0) {
+        Math_StepToF(&this->actor.world.pos.x, this->actor.home.pos.x, 2.0f);
+        Math_StepToF(&this->actor.world.pos.z, this->actor.home.pos.z, 2.0f);
+    }
+    if (this->actor.home.rot.y == this->actor.world.rot.y) {
+        func_808AFF9C(this);
+    }
+}
+
+void func_808B057C(EnAm* this) {
+    this->unk_240 = 6.0f;
+    this->unk_23C = Actor_YawToPoint(&this->actor, &this->actor.home.pos);
+    this->unk_236 = 1;
+    this->actionFunc = func_808B05C8;
+}
+
+void func_808B05C8(EnAm* this, GlobalContext* globalCtx) {
+    this->unk_23C = Actor_YawToPoint(&this->actor, &this->actor.home.pos);
+    func_808B0208(this, globalCtx);
+    if (Actor_XZDistanceToPoint(&this->actor, &this->actor.home.pos) < 8.0f) {
+        func_808B04E4(this);
+    }
+}
+
+void func_808B0640(EnAm* this) {
+    this->unk_236 = 0x28;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    this->unk_240 = 0.0f;
+    this->actionFunc = func_808B066C;
+}
+
+void func_808B066C(EnAm* this, GlobalContext* globalCtx) {
+    if (this->unk_236 != 0) {
+        this->unk_236--;
+    } else {
+        this->unk_23C = this->actor.yawTowardsPlayer;
+        func_808B0208(this, globalCtx);
+        if (this->unk_23C == this->actor.shape.rot.y) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_AMOS_VOICE);
+            func_808B0358(this);
+        }
+    }
+}
+
+void func_808B06D0(EnAm* this, GlobalContext* globalCtx) {
+    Animation_Change(&this->skelAnime, &D_06005B3C, 1.0f, 4.0f, (Animation_GetLastFrame(&D_06005B3C) - 6), 2, 0.0f);
+    func_800BE504(&this->actor, &this->enemyCollider);
+    this->actor.speedXZ = 6.0f;
+    func_800BCB70(&this->actor, 0x4000, 0xFF, 0, (Animation_GetLastFrame(&D_06005B3C) - 0xA));
```

Comment:
```suggestion
    func_800BCB70(&this->actor, 0x4000, 0xFF, 0, (Animation_GetLastFrame(&D_06005B3C) - 10));
```
AFAIK, frame counts are in decimal.

Reply:
It also looks like there's an extra set of brackets around Animation_GetLastFrame() - 10.